### PR TITLE
[1375] Resize the node to always display the full label

### LINKED ISSUE #1375: Resize the node with the minimum width of its inner Label
* [X] **I have checked that this feature has not yet been suggested by someone else.**

### Current behavior
Currently, if the node has been manually resized and when the label is changed, the node is never resized.

### Expected behavior
Even if the node has been manually resized, if the label is an inner label 
when the label is changed, the node width will have the minimum size to fully display its label.
...

### Cuttings backs

...

### Potential side effects

...

### Describe alternatives you've considered

* Label with line return
* Label with ellipse
Both cases are too complex to be managed rapidly.
A dedicated enhancement with a spec should be define in the future to better handle this scenario

### PR DESCRIPTION
**Pull request by lfasani**
Even if the node has been manually resized, if the label is an inner label, when the label is changed, the node will have the minimum width to fully display its label.

Bug: https://github.com/eclipse-sirius/sirius-components/issues/1375
Signed-off-by: Laurent Fasani <laurent.fasani@obeo.fr>

# Pull request template

## General purpose
What is the main goal of this pull request?
- [X] Bug fixes
- [ ] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [x] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [x] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [x] Have the relevant issues been added to the pull request?
- [x] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [x] Have the relevant issues been added to the same project and milestone as the pull request?
- [x] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: packages/diagrams/backend/sirius-components-diagrams-layout/src/main/java/org/eclipse/sirius/components/diagrams/layout/incremental/updater/ContainmentUpdater.java [hunk 1]
```diff
@@ -86,17 +86,31 @@ public void update(IContainerLayoutData container) {
...
+        } else {
+            if (container instanceof NodeLayoutData) {
+                NodeLayoutData nodeLayoutData = (NodeLayoutData) container;
+                shouldUpdate = !(nodeLayoutData.isResizedByUser() && this.hasEveryChildWithinItsBounds(nodeLayoutData));
+            }
 
-        if (container instanceof NodeLayoutData) {
-            NodeLayoutData nodeLayoutData = (NodeLayoutData) container;
-            shouldUpdate = !(nodeLayoutData.isResizedByUser() && this.hasEveryChildWithinItsBounds(nodeLayoutData));
+            shouldUpdate = shouldUpdate && (!container.getChildrenNodes().isEmpty() || isLabelContained);
```
Comment (gcoutable) [inline packages/diagrams/backend/sirius-components-diagrams-layout/src/main/java/org/eclipse/sirius/components/diagrams/layout/incremental/updater/ContainmentUpdater.java]: I barely understand when the node should be updated or not. I think the `if .. then ..` can be flattened.

**Review by gcoutable — CHANGES_REQUESTED** [review]

File: packages/diagrams/backend/sirius-components-diagrams-layout/src/main/java/org/eclipse/sirius/components/diagrams/layout/incremental/updater/ContainmentUpdater.java (hunk 1 above)
Comment (gcoutable) [reply packages/diagrams/backend/sirius-components-diagrams-layout/src/main/java/org/eclipse/sirius/components/diagrams/layout/incremental/updater/ContainmentUpdater.java]: Well, no matter what we do, this is barely understandable. Let's keep it that way.

**Review by gcoutable — CHANGES_REQUESTED** [review]
This does not work as intended. With this PR if the node is resized because of the label, the _node height_ is also updated while we only expect the _node width_ to be updated.

**Comment by sbegaudeau** [comment]
Closing this small fix since we will move into another direction with the support for label wrap.

**Pull request closed without merging**